andela/ah-bird-box
# #163388251 Add Social authentication via google and facebook

#### What does this PR do?
Adds backend implementation of social login with google and facebook

#### Description of Task to be completed?
> Add social auth test
> Enable google login
> Enable facebook login

#### How should this be manually tested?
_1. Ensure you have the below exports in your environment. The sample data provided can be used._
`export FACEBOOK_KEY=<facebook_key_value>`
`export FACEBOOK_SECRET=<facebook_secret>`
`export GOOGLE_OAUTH2_KEY=<google_auth_key_value>`
`export GOOGLE_OAUTH2_SECRET=<google_auth_secret>`
`export OAUTH2_ACCESS_TOKEN= <facebook_long_term_auth_token>`
_2. Run server_
_3. At postman, make a `post` request to ```http://127.0.0.1:8000/api/social/login/```_ with the body having the sample data below:
 **facebook**
`{
    "provider":"facebook",
     "access_token":"<OAUTH2_ACCESS_TOKEN_GOOGLE>"
}`
**google**
`{
    "provider":"google-oauth2",
     "access_token":"<OAUTH2_ACCESS_TOKEN_GOOGLE>"
}`
#### Any background context you want to provide?
Use `https://developers.google.com/oauthplayground` to generate your own google access token
#### What are the relevant pivotal tracker stories?
[#163388251](https://www.pivotaltracker.com/story/show/163388251)
#### Screenshots (if appropriate)
**Google**
![image](https://user-images.githubusercontent.com/29451552/52371840-99b0a700-2a67-11e9-96c2-080bf3e0621a.png)
**Facebook**
![image](https://user-images.githubusercontent.com/29451552/52372007-06c43c80-2a68-11e9-94b4-1a1f01a9ab98.png)
